src/commands/__tests__/undo-redo.ts

```diff
@@ -191,6 +196,130 @@ describe('undo', () => {
 
     expect(cursorThoughts).toMatchObject(expectedCursor)
   })
+
+  it('undo should restore all thoughts after a multicursor moveThoughtDown operation', () => {
+    store.dispatch([
+      importText({
+        text: `
+        - a
+        - b
+        - c
+        - d
+        - e`,
+      }),
+      setCursor(['a']),
+      addMulticursor(['a']),
+      addMulticursor(['b']),
+      addMulticursor(['c']),
+    ])
+
+    // Execute moveThoughtDown on all selected thoughts
+    executeCommandWithMulticursor(moveThoughtDownCommand, { store })
+
+    // Check intermediate state after moveThoughtDown but before undo
+    // This verifies that all three thoughts were moved down correctly
+    let exported = exportContext(store.getState(), [HOME_TOKEN], 'text/plain')
+    let expectedOutput = `- ${HOME_TOKEN}
+  - d
+  - a
+  - b
+  - c
+  - e`
+    expect(exported).toEqual(expectedOutput)
+
+    // Now perform one undo operation
+    store.dispatch(undo())
+
+    // On main branch, this would only move thought 'c' back up,
+    // leaving 'a' and 'b' in their moved positions
+    // With our fix, all three thoughts should be restored
```

Comment:
We usually want to avoid referencing branches or old versions of `main`, as this comment will be invalidated as soon as this PR is merged.

The title of the test already describes this well: "undo should restore all thoughts after a multicursor moveThoughtDown operation"